### PR TITLE
Sprite scale9

### PR DESCRIPTION
**Pull request by ricardoquesada**
Adds support for "sliced 9" sprite to `Sprite`.  This will deprecate `Scale9Sprite`.

It adds two new APIs to `Sprite`:
- `setCenterRect(Rect())`  // in pixel coordinates
- `setCenterRectNormalized(Rect())`   // in normalized coordinates (from 0 to 1)

To stretch the sprite, just call `sprite->setContentSize()`.
`setContentSize()` will stretch the sprite both in 9-slice mode, or "normal" mode.

### CODE REVIEW

File: cocos/2d/CCAutoPolygon.h [hunk 1]
```diff
@@ -81,6 +81,15 @@ class CC_DLL PolygonInfo
     void setQuad(V3F_C4B_T2F_Quad *quad);
 
     /**
+     * set the data to be a pointer to a number of Quads
+     * the member verts will not be released when this PolygonInfo destructs
+     * as the verts memory are managed by other objects
+     * @param quad  a pointer to the V3F_C4B_T2F_Quad quads
+     */
+    void setQuads(V3F_C4B_T2F_Quad *quads, int nubmerOfQuads);
```
Comment (dumganhar) [inline cocos/2d/CCAutoPolygon.h]: typo: nubmerOfQuads -> numberOfQuads

Comment (ricardoquesada) [reply cocos/2d/CCAutoPolygon.h]: fixed. thanks.

File: cocos/2d/CCAutoPolygon.cpp [hunk 2]
```diff
@@ -36,6 +36,18 @@ THE SOFTWARE.
...
+    0+4*0,1+4*0,2+4*0, 3+4*0,2+4*0,1+4*0,
+    0+4*1,1+4*1,2+4*1, 3+4*1,2+4*1,1+4*1,
+    0+4*2,1+4*2,2+4*2, 3+4*2,2+4*2,1+4*2,
+    0+4*3,1+4*3,2+4*3, 3+4*3,2+4*3,1+4*3,
+    0+4*4,1+4*4,2+4*4, 3+4*4,2+4*4,1+4*4,
+    0+4*5,1+4*5,2+4*5, 3+4*5,2+4*5,1+4*5,
+    0+4*6,1+4*6,2+4*6, 3+4*6,2+4*6,1+4*6,
+    0+4*7,1+4*7,2+4*7, 3+4*7,2+4*7,1+4*7,
+    0+4*8,1+4*8,2+4*8, 3+4*8,2+4*8,1+4*8,
+};
```
Comment (minggo) [inline cocos/2d/CCAutoPolygon.cpp:48]: Does it have 36 vertexes? I think it only has 16 vertexes? Am i wrong?

File: cocos/2d/CCSprite.cpp [hunk 3]
```diff
@@ -410,50 +416,114 @@ void Sprite::setTextureRect(const Rect& rect, bool rotated, const Size& untrimme
...
 
-    setContentSize(untrimmedSize);
-    setVertexRect(rect);
-    setTextureCoords(rect);
+    Node::setContentSize(untrimmedSize);
+    _originalContentSize = untrimmedSize;
 
-    float relativeOffsetX = _unflippedOffsetPositionFromCenter.x;
-    float relativeOffsetY = _unflippedOffsetPositionFromCenter.y;
+    // overrides might update _rect. Use _rect instead of _rect starting from here
```
Comment (minggo) [inline cocos/2d/CCSprite.cpp]: > Use _rect instead of _rect

What does it mean?

File: cocos/2d/CCAutoPolygon.cpp (hunk 2 above)
Comment (ricardoquesada) [reply cocos/2d/CCAutoPolygon.cpp:48]: It has the indices for 9 quads, and each quad requires 6 indices and 4 vertices, so it needs: 9 \* 6 = 54 indices and 9*4=36 vertices.

File: cocos/2d/CCSprite.cpp (hunk 3 above)
Comment (ricardoquesada) [inline cocos/2d/CCSprite.cpp]: bug: it should read: "Use _rect instead of rect"

Comment (ricardoquesada) [reply cocos/2d/CCSprite.cpp]: deprecated comment. I'll remove it.

File: cocos/2d/CCAutoPolygon.cpp (hunk 2 above)
Comment (dumganhar) [reply cocos/2d/CCAutoPolygon.cpp:48]: scale9sprite

I guess there are only 16 vertices and 4 x 9 = 36 indices (one quad requires 4 indices, there are 9 quads, so there are 4 x 9 = 36 indices). @ricardoquesada , anything I missed?

Comment (ricardoquesada) [reply cocos/2d/CCAutoPolygon.cpp:48]: you are right. I'm just reusing the Sprite's quad logic.

**Comment by ricardoquesada** [comment]
@minggo @dumganhar 
are you Ok if I merge this PR? or are you still reviewing ?

The thing that you mentioned about the Triangles vs. Quad is valid. I'll try to implement after I finish the Creator for C++. 

thanks.

**Comment by dumganhar** [comment]
@ricardoquesada , it looks good for me. 
@minggo , did you finish the reviewing?

**Comment by ricardoquesada** [comment]
@dumganhar thanks. I've just added a few more fixes.

@minggo any update? thanks.

**Comment by minggo** [comment]
@ricardoquesada sorry for late reply. Looks good to me.

**Comment by ricardoquesada** [comment]
@minggo @dumganhar ok. thanks. I'll merge it tomorrow (Tuesday)

**Pull request merged**